pep-0XXX-environment-markers.rst

```diff
@@ -0,0 +1,184 @@
+Post-History: 03-Jul-2015
+Replaces: 0345 (part of)
+
+
+Abstract
+========
+
+An **environment marker** describes a condition about the current execution
+environment. They are used to indicate when certain dependencies are only
+required in particular environments, and to indicate supported platforms
+for distributions with additional constraints beyond the availability of a
+Python runtime.
+
+Environment markers were first specified in PEP-0345[1]. PEP-0426[2] (which
+would replace PEP-0345) proposed extensions to the markers. When
+2.7.10 was released, even these extensions became insufficient due to
+their reliance on simple lexical comparisons, and thus this PEP has
+been born.
+
+
+Rationale
+=========
+
+Many Python packages are written with portability in mind.
+
+For many packages this means they aim to support a wide range of
+Python releases. If they depend on libraries such as argparse or
+simplejson - which started as external libraries, but later got
+incorporated into core - specifying a single set of requirements is
+difficult, as the set of required packages differs depending on the
+version of Python in use.
+
+For other packages, this means broad cross-platform support. However,
+this may mean (for instance) that certain packages are required for
+installation on Windows, but not other platforms.
+
+Environment Markers attempt to provide more flexibility in a list of
+requirements by allowing the developer to give some information about
+the conditions under which a requirement should be installed.
+
+
+Examples
+========
+
+Here are some examples of such markers::
+
+   "sys_platform == 'win32'"
+   "platform_machine == 'i386'"
+   "python_version == '2.4' or python_version == '2.5'"
+   "python_version>='2.7' and python_version<'3.0'"
+   "'linux' in sys_platform"
+
+And here's an example of some conditional metadata for a distribution that
+requires PyWin32 both at runtime and buildtime when using Windows::
+
+    "name": "ComfyChair",
+    "run_requires": [
+      {
+        "requires": ["pywin32 > 1.0"],
+        "environment": "sys.platform == 'win32'"
+      }
+    ]
+    "build_requires": [
+      {
+        "requires": ["pywin32 > 1.0"],
+        "environment": "sys.platform == 'win32'"
+      }
+    ]
+
+
+Micro-language
+==============
+
+The micro-language behind this is a simple subset of Python. It compares::
+
+* strings with the ``==`` and ``in`` operators (and their opposites)
+* version numbers with the ``<``, ``<=``, ``>=``, and ``<`` operators
+  in addition to those supported for strings
+
+The usual boolen operators ``and`` and ``or`` can be used to combine
+expressions, and parentheses are supported for grouping.
+
+The pseudo-grammar is not ::
+
+    MARKER: EXPR [(and|or) EXPR]*
+    EXPR: ("(" MARKER ")") | (SUBEXPR [CMPOP SUBEXPR])
+    CMPOP: (==|!=|<|>|<=|>=|in|not in)
+
+but my memory of BNF isn't quite up to fixing it just now.
+
+``SUBEXPR`` is either a Python string (such as ``'2.4'``, or
```

Comment:
In this case, the main argument for the quotes is "that's what PEP 345 used, and we're aiming to standardise and clarify an existing practice, rather than change it".

The history independent rationalisation is that it helps create a greater visual distinction from the preceding version specifier in the "specifier; marker" syntax that both pip and setuptools use.